src/App/Pages/CaptchaProtectedViewModel.cs

```diff
@@ -16,6 +16,23 @@ public abstract class CaptchaProtectedViewModel : BaseViewModel
         protected abstract IPlatformUtilsService platformUtilsService { get; }
         protected string _captchaToken = null;
 
+        protected async Task<bool> HandleCaptchaAsync(string captchaSiteKey, bool needsCaptcha, Func<Task> onSuccess)
+        {
+            if (!needsCaptcha)
+            {
+                _captchaToken = null;
+                return true;
+            }
+
+            if(await HandleCaptchaAsync(captchaSiteKey))
+            {
+                await onSuccess();
+                _captchaToken = null;
+                return false;
+            }
+            return false;
+        }
```

Comment:
IMO, given the name, it should return `true` if the captcha is needed and `false` if not.
Also, given that the last two `return` are the same then you only need the last one.